src/lavinmq/http/controller/queues.cr

```diff
@@ -41,6 +42,21 @@ module LavinMQ
           end
         end
 
+        get "/api/queues/:vhost/:name/unacked" do |context, params|
+          with_vhost(context, params) do |vhost|
+            refuse_unless_management(context, user(context), vhost)
+            unacked_messages = Hash(UInt64, UnackedMessage).new
+            @amqp_server.vhosts[vhost].queues[params["name"]].@consumers.each do |c|
+              c.@channel.@unacked.reject { |u| u.queue.name != params["name"] }.each do |u|
```

Comment:
`reject` generates a new array, but `each.reject` will "stream" the result, without generating a new array. 

```suggestion
              c.@channel.@unacked.each.select { |u| u.queue.name == params["name"] } do |u|
```

Reply:
and accessing instance variables with `@` in a class should be avoided. Instead we can expose a `unacked` method on `Consumer` that does this.